src/js/components/molecules/list-section/list-section.ts

```diff
@@ -1,51 +1,91 @@
 
-import { ListSection as IListSection } from '@ultimaker/ultimaker.com-model-definitions/dist/molecules/sections/ListSection';
-import Events from 'constants/events';
-import WithRender from './list-section.vue.html';
-
 @WithRender
 @Component({
     name: 'ListSection',
 })
-export class ListSection extends Mixins(StepRowAnimation) implements IListSection {
-    @Prop({ type: Array, required: true }) cards!: IListSection['cards'];
-    @Prop({ type: Object }) limit?: IListSection['limit'];
-    @Prop({ type: String }) title?: IListSection['title'];
-    @Prop({ type: Object }) tooltip?: IListSection['tooltip'];
-
-    viewportUtil: ViewportUtil = new ViewportUtil();
-    visibleTooltip: boolean = false;
-
-    $route;
+export class ListSection extends Mixins(StepRowAnimation) implements ListSectionProps {
+    @Prop({ type: Array, required: true }) cards!: ListSectionProps['cards'];
+    @Prop({ type: Array }) filterCategories?: ListSectionProps['filterCategories'];
+    @Prop({ type: Object }) limit?: ListSectionProps['limit'];
+    @Prop({ type: String }) title?: ListSectionProps['title'];
+    @Prop({ type: Object }) tooltip?: ListSectionProps['tooltip'];
+
+    /**
+     * @type PublicEventService.emit
+     */
     $emitPublic;
 
-    showMax: number = 6;
-    expanded: boolean = false;
+    $route;
 
     chunks: object[] = [];
     chunkSize: number = 3;
     chunkIndex: number = 0;
+    expanded: boolean = false;
+    filters: object = {};
+    filtersAll: any = [];
+    showMax: number = 6;
+    viewportUtil: ViewportUtil = new ViewportUtil();
     visibleChunks: number = 0;
+    visibleTooltip: boolean = false;
+
+    beforeDestroy(): void {
+        this.viewportUtil.removeResizeHandler(this.handleResize);
+    }
+
+    async mounted() {
+        if (!BrowserCapabilities.isBrowser) {
+            return;
+        }
+
+        await this.viewportUtil.addResizeHandler(this.handleResize);
+        await this.viewportUtil.triggerResize();
+        await this.handleResize();
+
+        this.createChunks();
+    }
+
+    get filteredCards() {
+        if (this.filtersAll.length) {
+            // @ts-ignore
+            return this.cards.filter((card) => card.labels.some((label) => this.filtersAll.includes(label)));
+        }
+
+        return this.cards;
+    }
 
     @Watch('cards')
     createChunks() {
         this.chunkIndex = 0;
         this.chunks = [];
 
-        while (this.chunkIndex < this.cards.length) {
-            this.chunks.push(this.cards.slice(this.chunkIndex, this.chunkSize + this.chunkIndex));
+        while (this.chunkIndex < this.filteredCards.length) {
+            this.chunks.push(this.filteredCards.slice(this.chunkIndex, this.chunkSize + this.chunkIndex));
             this.chunkIndex += this.chunkSize;
         }
     }
 
+    handleFilterCategoryChange(payload): void {
+        this.filters[payload.category] = payload.filters;
```

Comment:
Maybe I'm not quite getting the logic behind this. But why put the filters into the category and then loop them again. Why not ```this.filtersAll = [ ... this.filtersAll, ... payload.filters]```